### ISSUE TITLE
Odd history behaviour in ssh + psql

### ISSUE THREAD

**Issue by pnomolos** [gnunn1/tilix#1121]
Unfortunately I don't have any tools for capturing GIFs on Linux, so here's a few screenshots with an explanation:

Initial state
![](https://cl.ly/2j003S121402/Tilix%3A%20Default_005.png)

Pressed <Up Arrow> once to retrieve last history entry.  It's actually a multi-line SQL query but it's all being jammed onto a single line.
![](https://cl.ly/120N3P2R1D2r/Tilix%3A%20Default_006.png)

Pressed <Up Arrow> one more time to retrieve the second-to-last history entry.  "\e" is the correct entry, but it's removed the last 3 lines of output.
![](https://cl.ly/1v0D0s39403i/Tilix%3A%20Default_008.png)

If I alternate between pressing up twice, then down twice, it'll eventually erase all the content on the current Tilix screen (won't start scrolling back up or anything weird, though). 

- The standard Ubuntu 17.04 terminal doesn't exhibit this behaviour.
- I wasn't able to reproduce this on my local machine even with the same query.  Had to be across an SSH session.

**Comment by gnunn1** [comment]
Tilix uses the same VTE widget as gnome-terminal so there should not be any difference at the terminal level. Having said that, I would suggest to try launching tilix, with no other instance running, from the command line in gnome-terminal. 

Tilix does use dbus activation and this results in a somewhat different environment in terms of environment variables. Launching it from the command line eliminates this. 

Let me know if you see any difference in behavior.

**Comment by pnomolos** [comment]
@gnunn1 I think I've figured out what the culprit is, and it _is_ reproducible in gnome-terminal.  That being said, perhaps you'll have some insight into what I can do to fix it (if anything).  The problem was that I would start an edit session within psql, then resize the window while my editor (vi, if that matters) was open.  Once I dropped back into the psql session things were all off, presumably due to a mis-match somewhere in terms of what size the term thinks it is.

Feel free to close, as I don't believe this is a Tilix issue.
